pkg/cluster/components/apiserver.go

```diff
@@ -212,5 +213,10 @@ func NewAPIServerStaticPod(cfg *config.ControlPlaneConfiguration) (*corev1.Pod,
 	if err := appendExtraVolumes(p, cfg.KubeAPIServerConfiguration.ExtraVolumes); err != nil {
 		return nil, err
 	}
+
+	if err := appendExtraLabels(p, cfg.KubeAPIServerConfiguration.ExtraLabels); err != nil {
```

Comment:
this function can never return an error, right? so we can drop all of this checking/logging